cargo-near/src/util/mod.rs

```diff
@@ -241,8 +243,14 @@ pub(crate) fn compile_project(
 /// Create the directory if it doesn't exist, and return the absolute path to it.
 pub(crate) fn force_canonicalize_dir(dir: &Utf8Path) -> color_eyre::eyre::Result<Utf8PathBuf> {
     fs::create_dir_all(dir).wrap_err_with(|| format!("failed to create directory `{}`", dir))?;
-    dir.canonicalize_utf8()
-        .wrap_err_with(|| format!("failed to access output directory `{}`", dir))
+    // use canonicalize from `dunce` create instead of default one from std because it's compatible with Windows UNC paths
+    // and don't breake cargo compilation on Windows
+    // https://github.com/rust-lang/rust/issues/42869
+    let compatible_path = dunce::canonicalize(dir);
+    match compatible_path {
+        Ok(path) => Ok(Utf8PathBuf::from_path_buf(path).unwrap()),
+        Err(err) => Err(err).wrap_err_with(|| format!("failed to canonicalize path: {} ", dir)),
+    }
```

Comment:
We should not unwrap, so handle the error gracefully, and you shouldn't need to match explicitly, instead, use `?` operator:

```suggestion
    Utf8PathBuf::from_path_buf(
        dunce::canonicalize(dir)
            .wrap_err_with(|| format!("failed to canonicalize path: {} ", dir))?
    )
    .wrap_err_with(|| format!("failed to convert canonicalized path to UTF-8 path: {} ", dir))
    .into()
```